### PR TITLE
Geometry approximation example

### PR DESCRIPTION
**Pull request by emmwel**
Includes a simplified python script and .md file for the TTK Examples website

### CODE REVIEW

File: python/geometryApproximation.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+threshold1.Scalars = ['CELLS', 'TriangleDistortion']
+threshold1.ThresholdRange = [0.0, 0.02]
+
+# create a new 'TTK IcospheresFromPoints'
+tTKIcospheresFromPoints1 = TTKIcospheresFromPoints(Input=tTKIcosphereFromObject1)
+tTKIcospheresFromPoints1.Subdivisions = 1
+tTKIcospheresFromPoints1.Radius = 0.01
+tTKIcospheresFromPoints1.ComputeNormals = 0
+
+SaveData('CameraNodes.vtp', tTKIcospheresFromPoints1)
```
Comment (julien-tierny) [inline python/geometryApproximation.py]: I don't believe this output needs to be stored for this example.
(that would simplify the script)

File: python/geometryApproximation.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+tTKDepthImageBasedGeometryApproximation1 = TTKDepthImageBasedGeometryApproximation(Input=tTKCinemaImaging1)
+tTKDepthImageBasedGeometryApproximation1.DepthArray = ['POINTS', 'Depth']
+
+# create a new 'Threshold'
+threshold1 = Threshold(Input=tTKDepthImageBasedGeometryApproximation1)
+threshold1.Scalars = ['CELLS', 'TriangleDistortion']
+threshold1.ThresholdRange = [0.0, 0.02]
+
+# create a new 'TTK IcospheresFromPoints'
+tTKIcospheresFromPoints1 = TTKIcospheresFromPoints(Input=tTKIcosphereFromObject1)
```
Comment (julien-tierny) [inline python/geometryApproximation.py]: this pipeline node can be removed (see comment below)

File: python/geometryApproximation.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env python
+
+from paraview.simple import *
+
+# create a new 'XML PolyData Reader'
+streamlinesvtp = XMLPolyDataReader(FileName=['/home/emma/ttk/ttk-source/ttk-data-emmwel/GroundWater.cdb/streamlines.vtp'])
```
Comment (julien-tierny) [inline python/geometryApproximation.py]: I don't believe the streamlines need to be loaded in this srcipt.
(they are not used further down the script)

File: docs/geometryApproximation.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
+# Geometry Approximation example
+
+![Geometry Approximation Example Image](https://topology-tool-kit.github.io/img/gallery/geometryApproximation.jpg)
+
+## Pipeline description
+This example loads two geometric structures: the streamlines of a section of ground water and its surrounding stone.
```
Comment (julien-tierny) [inline docs/geometryApproximation.md:6]: I don't believe the python script needs to load the streamlines (see my comments below)

File: python/geometryApproximation.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env python
+
+from paraview.simple import *
+
+# create a new 'XML PolyData Reader'
+streamlinesvtp = XMLPolyDataReader(FileName=['/home/emma/ttk/ttk-source/ttk-data-emmwel/GroundWater.cdb/streamlines.vtp'])
+
+# create a new 'XML PolyData Reader'
+stonevtp = XMLPolyDataReader(FileName=['/home/emma/ttk/ttk-source/ttk-data-emmwel/GroundWater.cdb/stone.vtp'])
```
Comment (julien-tierny) [inline python/geometryApproximation.py]: thanks for changing from an absolute to a relative path!
(i.e. just `GroundWater.cdb/stone.vtp`)

File: docs/geometryApproximation.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+A fake shadow is put on the stone mesh to give the mesh depth values. The stone object is placed within a sphere using [IcosphereFromObject](https://topology-tool-kit.github.io/doc/html/classttkIcosphereFromObject.html). The object and the icosphere is then given to the [CinemaImaging](https://topology-tool-kit.github.io/doc/html/classttkCinemaImaging.html) filter. It generates multiple images of the stone object, and the images are taken from cameras placed on the vertices of the icosphere.
+
+ An approximation of the geometry is calculated from the images using [DepthImageBasedGeometryApproximation](https://topology-tool-kit.github.io/doc/html/classttkDepthImageBasedGeometryApproximation.html). The approximated geometry can be extracted using the Threshold filter.
```
Comment (julien-tierny) [inline docs/geometryApproximation.md]: Threshold -› `Threshold`

File: docs/geometryApproximation.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+Giving the icosphere around the object to [IcospheresFromPoints](https://topology-tool-kit.github.io/doc/html/classttkIcospheresFromPoints.html) yields icospheres at the coordinates of the cameras.
```
Comment (julien-tierny) [inline docs/geometryApproximation.md]: I believe that part of the script can be removed.

File: docs/geometryApproximation.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+```
+
+## Python code
+
+``` python  linenums="1"
+--8<-- "python/geometryApproximation.py"
+```
+
+## Inputs
+- [streamlines.vtp](https://github.com/topology-tool-kit/ttk-data/tree/dev/GroundWater.cdb/streamlines.vtp): a mesh of the streamlines of the ground water.
```
Comment (julien-tierny) [inline docs/geometryApproximation.md]: I'd suggest to remove that input as it is not used in the script

File: docs/geometryApproximation.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+- [stone.vtp](https://github.com/topology-tool-kit/ttk-data/tree/dev/GroundWater.cdb/stone.vtp): a mesh of the stone around the ground water.
+
+## Outputs
+- `CameraNodes.vtp`: the points representing the positions of the cameras used to take images of the geometry.
```
Comment (julien-tierny) [inline docs/geometryApproximation.md]: I'd suggest to remove that output

**Review by julien-tierny — CHANGES_REQUESTED** [review]
thanks a lot for the PR.
could you please address the few minor remarks I entered?
thanks!

**Comment by julien-tierny** [comment]
hi @emmwel 
this is a friendly reminder: if you could address my quick remarks, that'd be awesome.
thanks!

**Comment by emmwel** [comment]
> hi @emmwel this is a friendly reminder: if you could address my quick remarks, that'd be awesome. thanks!

Hi sorry for the late fix, I did it now and hope I got everything :)

**Comment by julien-tierny** [comment]
alright, looks good. thanks!

**Comment by julien-tierny** [comment]
can you please PR ttk's source code to include pointers to this example in the corresponding Doxygen and ParaView documentations?
thanks!

**Pull request merged**